File: public/examples/integrations/toolkits/dropbox/search_files_and_folders_example_call_tool.py
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

TOOL_NAME = "Dropbox.SearchFilesAndFolders"

auth_response = client.tools.authorize(tool_name=TOOL_NAME)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    "keywords": "quarterly report 2025",
    "search_in_folder_path": "/My Documents/My Folder",
    "filter_by_category": ["pdf", "document"],
    "limit": 5,
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
)
print(response)

File: public/examples/integrations/toolkits/dropbox/search_files_and_folders_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const TOOL_NAME = "Dropbox.SearchFilesAndFolders";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  keywords: "quarterly report 2025",
  search_in_folder_path: "/My Documents/My Folder",
  filter_by_category: ["pdf", "document"],
  limit: 5,
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
});

console.log(response);
